### PR TITLE
[TIMOB-23496] TiAPI Fix: Converge on same Object properties/types for…

### PR DESCRIPTION
**Pull request by maggieaxway**
… GeocodedAddress

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23496

### CODE REVIEW

File: android/modules/geolocation/src/java/ti/modules/titanium/geolocation/TiLocation.java [hunk 1]
```diff
@@ -335,13 +335,12 @@ private KrollDict buildAddress(JSONObject place)
 		address.put(TiC.PROPERTY_CITY, place.optString(TiC.PROPERTY_CITY, ""));
 		address.put(TiC.PROPERTY_REGION1, ""); // AdminArea
 		address.put(TiC.PROPERTY_REGION2, ""); // SubAdminArea
-		address.put(TiC.PROPERTY_POSTAL_CODE, place.optString("zipcode", ""));
+		address.put(TiC.PROPERTY_ZIPCODE, place.optString(TiC.PROPERTY_ZIPCODE, ""));
```
Comment (hansemannn) [inline android/modules/geolocation/src/java/ti/modules/titanium/geolocation/TiLocation.java]: This is a breaking change. We need to retain the old values until the next major release.

File: android/modules/geolocation/src/java/ti/modules/titanium/geolocation/TiLocation.java [hunk 2]
```diff
@@ -335,13 +335,12 @@ private KrollDict buildAddress(JSONObject place)
...
 		address.put(TiC.PROPERTY_COUNTRY, place.optString(TiC.PROPERTY_COUNTRY, ""));
 		address.put(TiC.PROPERTY_STATE, place.optString(TiC.PROPERTY_STATE, ""));
-		address.put("countryCode", place.optString(TiC.PROPERTY_COUNTRY_CODE, "")); // TIMOB-4478, remove this later, was old android name
```
Comment (hansemannn) [inline android/modules/geolocation/src/java/ti/modules/titanium/geolocation/TiLocation.java]: Breaking change as well. `countryCode` != `country_code`.

File: apidoc/Titanium/Geolocation/Geolocation.yml [hunk 3]
```diff
@@ -1008,13 +1008,13 @@ properties:
...
     summary: Longitude of the geocoded address.
-    type: String
+    type: Number
     since: { mobileweb: "3.1.1" }
     platforms: [android, iphone, ipad, mobileweb]
 
   - name: latitude
     summary: Latitude of the geocoded address.
-    type: String
+    type: Number
```
Comment (hansemannn) [inline apidoc/Titanium/Geolocation/Geolocation.yml]: Need to leave a note in the description that is was unified in 6.1.0.

**Review by hansemannn — CHANGES_REQUESTED** [review]
Left some noted to the breaking changes this PR causes. We should retain all old values and remove the duplicate ones in a follow-up 7.0.0 ticket. 

Also, I removed the 6.0.2 version. This is no 6.0.0 regression, so it should go into the next feature release (6.1.0) instead.

**Comment by maggieaxway** [comment]
Test case:

var win = Ti.UI.createWindow({
  backgroundColor: 'white',
  exitOnClose: true,
  fullscreen: false,
  layout: 'vertical',
  title: 'TIMOB-23492'
});

var label1 = Ti.UI.createLabel({
  color: '#900',
  font: { fontSize:18 },
  textAlign: Ti.UI.TEXT_ALIGNMENT_LEFT,
  width: Ti.UI.SIZE, height: Ti.UI.SIZE
});

Ti.Geolocation.reverseGeocoder(1.278509, 103.848656, function (response) {

    var location = JSON.stringify(response, null, ' ');
    Ti.API.info(location);
    label1.setText(response.places[0].country_code);

});

win.add(label1);
win.open();

//====== Log output ======//

02-16 20:38:29.928 28645-28674/com.titanium.test I/TiAPI:  {
                                                           "source": {
                                                            "locationServicesEnabled": true,
                                                            "hasCompass": true,
                                                            "lastGeolocation": null,
                                                            "apiName": "Ti.Geolocation",
                                                            "bubbleParent": true,
                                                            "__propertiesDefined__": true
                                                           },
                                                           "code": 0,
                                                           "places": [
                                                            {
                                                             "address": "71 Robinson Road, 068895 Singapore, Singapore",
                                                             "street": "Robinson Road",
                                                             "country_code": "SG",
                                                             "country": "Singapore",
                                                             "street1": "Robinson Road",
                                                             "longitude": 103.848796,
                                                             "region2": "",
                                                             "zipcode": "068895",
                                                             "city": "Singapore",
                                                             "latitude": 1.2783746,
                                                             "region1": "",
                                                             "state": "",
                                                             "displayAddress": "71 Robinson Road, 068895 Singapore, Singapore"
                                                            }
                                                           ],
                                                           "success": true
                                                          }


//+++++++++++++++++++//

Related ticket: https://jira.appcelerator.org/browse/TIMOB-23492

Related docs: http://docs.appcelerator.com/platform/latest/#!/api/GeocodedAddress


Concerns: 

country_code (on all platforms), and countryCode (only on Android) (we should likely have the same property, and given our naming scheme, it should be called "countryCode")

If we change the property name to countryCode will affect most current users. The JSONObject name is country_code, that's why the TiC.java line 1195 is this:

	public static final String PROPERTY_COUNTRY_CODE = "country_code";

**Comment by garymathews** [comment]
Update the `TiC.PROPERTY_COUNTRY_CODE` constant to be `countryCode`. Lets also throw a warning about the removal of `postalCode` and `country_code`.

```Java
Log.w(TAG, "postalCode has been removed please use zipCode instead!");
Log.w(TAG, "country_code has been removed please use countryCode instead!");
```

**Comment by mukherjee2** [comment]
@maggieaxway can you please look at the merge conflicts? Is this ready for QE testing? It looks like it needs CR approval. @hansemannn , @garymathews are there still changes requested, or does CR pass?

**Comment by maggieaxway** [comment]
@hansemannn , @garymathews please don't review it yet. Couldn't solve the conflict due to delete repo accidentally in the past. Will clone to a new PR

**Comment by mukherjee2** [comment]
@maggieaxway can this be moved to 7.1.0? I don't think there's enough time for to resolve conflicts+CR+QE testing to make it in for release tomorrow.

**Comment by hansemannn** [comment]
@mukherjee2 Yes, but likely 8.0.0 then, because it looks like a breaking change.

**Comment by mukherjee2** [comment]
@hansemannn ok thanks. @eric34 can this be moved to 8.0? Seems that conflicts are not straightforward to resolve, and there are breaking changes which affects multiple platforms.

**Comment by sgtcoolguy** [comment]
I think this PR and ticket need to be revisited badly. We should try and get any deprecations into 7.x ASAP if we know we'll be changing the API here. I also suspect that we should be able to retain the existing properties as deprecated for now and introduce the new properties without necessarily causing breaking changes.

**Comment by hansemannn** [comment]
@sgtcoolguy Agree, let's try that for 7.0.0 GA. Scheduling for tomorrow.

**Comment by hansemannn** [comment]
Superseded by https://github.com/appcelerator/titanium_mobile/pull/9647.

**Pull request closed without merging**